src/java.base/share/classes/sun/nio/ch/DatagramChannelImpl.java

```diff
@@ -1732,11 +1738,21 @@ private void implCloseBlockingMode() throws IOException {
                 long reader = readerThread;
                 long writer = writerThread;
                 if (reader != 0 || writer != 0) {
-                    nd.preClose(fd);
-                    if (reader != 0)
-                        NativeThread.signal(reader);
-                    if (writer != 0)
-                        NativeThread.signal(writer);
+                    boolean isAix = System.getProperty("os.name").startsWith("AIX");
+                    if (isAix) {
+                        if (reader != 0)
+                            NativeThread.signal(reader);
+                        if (writer != 0)
+                            NativeThread.signal(writer);
+                        nd.preClose(fd);
+                    }
+                    else {
+                        nd.preClose(fd);
+                        if (reader != 0)
+                            NativeThread.signal(reader);
+                        if (writer != 0)
+                            NativeThread.signal(writer);
+                    }
```

Comment:
Rather than repeating as much code, suggest making the call to `preClose()` conditional before or after the `signal()` calls:
```java
                    if (!AIX.isAIX)
                        nd.preClose(fd);
                    if (reader != 0)
                        NativeThread.signal(reader);
                    if (writer != 0)
                        NativeThread.signal(writer);
                    if (AIX.isAIX)
                        nd.preClose(fd);
```